packages/react-keytips/src/docs/Spec.md

```diff
@@ -0,0 +1,190 @@
+
+### Fabric (v8)
+
+The [Keytips](https://developer.microsoft.com/en-us/fluentui#/controls/web/keytips) is present in Fabric v8.
+It provides `KeytipLayer` and `useKeytipRef` to manage the keytips.
+
+- [KeytipLayer](https://developer.microsoft.com/en-us/fluentui#/controls/web/keytips#IKeytipLayer) - Helps to register a
+  key sequence for entering and exiting from keytips, for going back to the previous level of keytips, provides necessary callbacks.
+- [useKeytipRef](https://github.com/microsoft/fluentui/blob/master/packages/react/src/components/KeytipData/useKeytipRef.ts) - Creates a ref,
+  which is used for passing to `Keytip` target element, handles setting necessary attributes.
+
+```tsx
+export default function App() {
+  // default is 'Alt-Windows' on Windows and 'Option-Control' on macOS
+  return <KeytipLayer content="Alt Windows" />;
+}
+```
+
+`useKeytipRef`:
+
+```tsx
+export const KeytipsRefExample = () => {
+  const checkboxRef = useKeytipRef<HTMLDivElement>({
+    keytipProps: {
+      id: 'CheckboxKeytip',
+      content: '1',
+      optionalProps: {
+        onExecute: (el: HTMLElement) => el.click(),
+      },
+    },
+  });
+
+  return <Checkbox label="checkbox" ref={checkboxRef} />;
+};
+```
+
+Some Fabric components have API inherited from it's previous versions, which was based on [keytipProps](https://developer.microsoft.com/en-us/fluentui#/controls/web/keytips#IKeytipProps) with `KeytipData` being used inside the component. This approach added extra bundle size to components and was replaced in `Link`, `Toggle`, `Checkbox`, `ComboBox`, `Dropdown`, and `SpinButton` components in favour of `useKeytipRef` hook API.
+
+```tsx
+export const KeytipsButtonExample = () => {
+  return (
+    <CompoundButton
+      text="CompoundButton"
+      onClick={handleOnClick}
+      keytipProps={{
+        id: 'CompoundButton',
+        content: '1B',
+        optionalProps: {
+          onExecute: (el: HTMLELement) => el.click();
+        },
+      }}
+    />
+  );
+}
+```
+
+### Northstar (v0)
+
+There is not prior implementation for this component in v0.
+
+## Sample Code
+
+```tsx
+import { Keytips, useKeytipRef, ExecuteKeytipEventHandler } from '@fluentui-contrib/react-keytips';
+
+const App = () => {
+  const onExecute: ExecuteKeytipEventHandler = (ev, ({ targetElement })) => {
+    targetElement.click();
+  };
+
+  const checkboxRef = useKeytipRef({
+    keySequences: ['a'],
+    content: ['A']
+    onExecute,
+  });
+
+  return (
+    <>
+      {/* Should be added once at the top level of your app */}
+      <Keytips />
+      <Checkbox label="checkbox" ref={checkboxRef} />
+    </>
+};
+```
+
+## Variants
+
+### Positioning Variants
+
+The `Keytips` component using [positioning API](https://react.fluentui.dev/?path=/docs/concepts-developer-positioning-components--docs) and can be controlled by `positioning` prop in `useKeytipRef` hook.
+The keytip is positioned below and centered to the target element by default.
```

Comment:
Maybe it'll be good to include other positioning variants too, not only default one